project/hsiehbingting/README.rst

```diff
@@ -0,0 +1,117 @@
+#################
+VisualizeWifi
+#################
+
+
+*****************
+Basic Information
+*****************
+* **Project Name**: VisualizeWifi
+* **Github Repo**: https://github.com/hsiehbingting/VisualizeWifi
+
+  
+*****************
+Problem to Solve
+*****************
+The project aims to assist in the design and construction of wireless
+network environments by using T-IPLM (Tata Indoor Path Loss Model)
+to predict and visualize indoor signal strength distribution.
+  
+Because T-IPLM takes into account the material and quantity of obstacles,
+T-IPLM has better accuracy than some popular path loss models such as 
+ITU-R and logarithmic distance.
+
+  
+*****************
+Prospective Users
+*****************
+Anyone who wants to set up or analyze an indoor 
+wireless network environment.
+
+
+*****************
+System Architecture
+*****************
+* **Input**: 
+    The user inputs room size, wall position and materials, and signal
+    source parameters in the Python interface.
+
+* **Compute**:
+    The C++ module uses the T-IPLM model to calculate the path loss at
+    various points in the room, based on the signal source location and
+    the room's structural properties (wall positions and materials).
+
+* **Analyze results**:
+    The C++ module computes the signal strength at each point by combining
+    the signal source strength with the previously calculated path loss.
+
+* **Visualize results**:
+    Use Matplotlib to visualize the signal distribution, generating 2D 
+    heatmap of the WiFi signal coverage.
+
+
+*****************
+API Description
+*****************
+
+Function in Python
++++++++++++++++++
+* ``add_room``: Define the room's size.
+
+* ``add_wall``: Add a wall to the room using the specified material.
+
+* ``set_signal_source``: Set the signal source's position and power.
+
+* ``visualize_signal_distribution``: Visualize the signal strength.
+
+
+Function in C++
++++++++++++++++++
+* ``compute_path_loss``: Compute the path loss at a given point.
+
+* ``compute_signal_strength``: calculates the signal strength at a given point.
+
+
+*****************
+Engineering Infrastructure
+*****************
+* Version control: Git
+
+* Build system: Makefile
+
+* Testing framework: Google Test and pytest
+
+* Continuous integration: Github actions
+
+
+*****************
+Schedule
+*****************
+1. Week 1-2 (9/30~):
+    * Set up the development environment.
+    * Familiarize with the T-IPLM model.
+    * Implement the Python interface for user input.
+    * Write unit tests and set up CI using GitHub Actions.
+
+2. Week 3-4 (10/14~):
+    * Implement C++ module for path loss calculation.
+
+3. Week 5-6 (10/28~):
+    * Optimize the C++ module with multithreading.
+    * Integrate Python and C++ using pybind11.
+
+4. week 7 (11/11~):
+    * Flexible Week: Adjustments, bug fixes, or any backlog from previous tasks.
+
+5. week 8-9 (11/18~):
+    * Write the visualization function in Python.
```

Comment:
2 weeks for visualization are too few.  How do you plan to visualize the results?  Please specify all graphical tool kits in your plan.